js/main.js

```diff
@@ -176,6 +203,25 @@ function getInfo(event) {
         infoViews.appendChild(renderCatDetailView(catData[i]));
         infoViews.className = 'info-view';
         breedsView.className = 'breeds-view hidden';
+        favView.className = 'fav-view hidden';
+      }
+    }
+  } else if (event.target.className === 'far fa-star') {
+    for (i = 0; i < catData.length; i++) {
+      if (catData[i].name === event.target.parentNode.parentNode.dataset.view) {
+        data.favorites.push(catData[i]);
+        favContainer.appendChild(renderCatFavItem(catData[i]));
+        event.target.className = 'fas fa-star';
+      }
+    }
+  } else if (event.target.className === 'cat-name fav') {
+    for (i = 0; i < catData.length; i++) {
+      if (event.target.parentNode.parentNode.dataset.view === catData[i].name) {
+        infoViews.textContent = '';
+        infoViews.appendChild(renderCatDetailView(catData[i]));
+        infoViews.className = 'info-view';
+        breedsView.className = 'breeds-view hidden';
+        favView.className = 'fav-view hidden';
       }
```

Comment:
Can these two loops be combined? It looks like they are looping over and searching for the same thing 🤔